content/docs/identity-providers/oidc.mdx

```diff
@@ -0,0 +1,204 @@
+---
+# cSpell:ignore keycloak, mykeycloak, myclient
+id: oidc
+title: OIDC
+description: Learn how integrate a generic OIDC provider with Pomerium.
+keywords: [keycloak, pomerium, oidc, generic oidc]
+lang: en-US
+---
+
+import SelfSignedCertWarning from '@site/content/docs/admonitions/_handle-self-signed-certificate-warning.mdx';
+
+# Generic OIDC
+
+This guide covers how to configure a generic OpenID Connect (**OIDC**) provider to work with Pomerium.
+
+## Background
+
+Pomerium provides default [identity provider settings](/docs/reference/identity-provider-settings) that allow you to seamlessly connect with a number of well-known identity provider (**IdP**) solutions. However, Pomerium can also connect with any generic OIDC provider if both Pomerium and the OIDC provider are properly configured.
+
+To learn how a generic [OIDC](https://openid.net/specs/openid-connect-core-1_0.html) implementation works with Pomerium, you will configure [Keycloak](https://www.keycloak.org/) as your OIDC provider.
+
+## Before you start
+
+This guide requires you to self-host the Keycloak server. To minimize set up time, you will deploy Pomerium, Keycloak, and the Pomerium Verify service in Docker containers.
+
+To complete this guide, you need:
+
+- [Docker](https://docs.docker.com/install/)
+- [Docker Compose](https://docs.docker.com/compose/install/)
+
+## Set up Docker Compose
+
+Create a `docker-compose.yaml` file and add the following configuration:
+
+```yaml title="docker-compose.yaml"
+version: '3'
+services:
+  mykeycloak:
+    image: quay.io/keycloak/keycloak:22.0.1
+    command:
+      - start-dev
+    environment:
+      - KEYCLOAK_ADMIN=admin
+      - KEYCLOAK_ADMIN_PASSWORD=admin
+    ports:
+      - 8080:8080
+    networks:
+      default:
+        aliases:
+          - keycloak.localhost.pomerium.io
+  pomerium:
+    image: pomerium/pomerium:v0.23.0
+    volumes:
+      - ./config.yaml:/pomerium/config.yaml:ro
+    ports:
+      - 443:443
+  verify:
+    image: pomerium/verify:latest
+    expose:
+      - 8000
+    environment:
+      JWKS_ENDPOINT: https://pomerium/.well-known/pomerium/jwks.json
+```
+
+Note the following points about this configuration:
+
+- `start-dev` runs the Keycloak server in Development mode
+- The admin console username and password are both `admin`
+- The Keycloak server is assigned a network alias (`keycloak.localhost.pomerium.io`) so Pomerium can generate a sign-in redirect URL that resolves to the Keycloak container on the host machine
+
+## Set up Keycloak
+
+In your terminal, run Docker Compose:
+
+```yaml
+docker compose up
+```
+
+When Keycloak is ready, navigate to `http://0.0.0.0:8080` to access the **Administration Console:**
```

Comment:
I'm not sure that all browsers will translate `0.0.0.0` to `localhost`. I'd recommend we specify `localhost` explicitly here instead.

I think we can also make this into a link?

```suggestion
When Keycloak is ready, navigate to [http://localhost:8080](http://localhost:8080) to access the **Administration Console:**
```

(and similarly for other occurrences of 0.0.0.0 in this page)